### PR TITLE
allow user to use custom function for simulation callback

### PR DESCRIPTION
**Pull request by AurelienJaquier**

### CODE REVIEW

**Review by Dinika — APPROVED** [review]

**Review by ilkilic — APPROVED** [review]

File: bluecellulab/simulation/simulation.py [hunk 1]
```diff
@@ -50,26 +51,29 @@ def init_progress_callback(self):
...
-                self.progress - 1), "/", " " * (100 - self.progress)))
-            sys.stdout.write("[%s%s%s]\n" % ("#" * (
-                self.progress - 1), ">", "." * (100 - self.progress)))
-            sys.stdout.write(" %s%s%s \n" % (" " * (
-                self.progress - 1), "\\", " " * (100 - self.progress)))
-        sys.stdout.flush()
+            self.custom_progress_function()
 
         neuron.h.cvode.event(
             neuron.h.t + self.progress_dt, self.progress_callback)
```
Comment (bilalesi) [inline bluecellulab/simulation/simulation.py:79]: what if the cvode is disabled 
for `synaptome simulation,` we have to disable the `cvode`,  so i don't know if the event will be triggered hence the progress_callback will work ?

**Review by bilalesi — APPROVED** [review]
@AurelienJaquier thanks for the PR, I left a comment regarding `cvode`

File: bluecellulab/simulation/simulation.py (hunk 1 above)
Comment (AurelienJaquier) [reply bluecellulab/simulation/simulation.py:79]: I have tested with cvode disabled, and the callback function is still called.

Comment (bilalesi) [reply bluecellulab/simulation/simulation.py:79]: okey thanks

**Review by bilalesi — APPROVED** [review]

**Pull request merged**